### DISCUSSION TITLE
Foreman vs Cockpit

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit#16188] [Q&A] [answered]
I came across the project [Foreman](https://www.theforeman.org/).
What is the difference between Foreman and Cockpit and why is Red Hat building two similar products?

I read that Foreman is the upstream version of Red Hat Satellite? What is the place of Cockpit in the Red Hats software landscape? Couldn't find any information about this.

**Comment by martinpitt** [comment, marked as answer]
Foreman/Satellite are meant to administer/install your hardware machine in bulk. At most you can run the same CLI command on a number of machines and eventually get the output. Cockpit is an interactive Linux session to work with and troubleshoot a *single* machine, with lots of pages what to do in particular. Thus the two go hand in hand -- Foreman even has a "Web console" button where you can "drill down" into a particular machine to get a Cockpit UI.

**Reply by Verhoeckx** [reply]
Ah, thanks for the clear answer :smiley: !!  I think I now understand the difference!
